elements/rh-table/rh-table.ts

```diff
@@ -0,0 +1,173 @@
+import { LitElement, html } from 'lit';
+import { customElement } from 'lit/decorators/custom-element.js';
+
+import { Logger } from '@patternfly/pfe-core/controllers/logger.js';
+
+import styles from './rh-table.css';
+import { RequestSortEvent, RhSortButton } from './rh-sort-button.js';
+
+/**
+ * A table is a container for displaying information. It allows a user to scan, examine, and compare large amounts of data.
+ *
+ * @summary Organizes and displays information from a data set
+ *
+ * @slot               - an HTML table
+ * @slot    summary    - a brief description of the data
+ * @csspart container  - table container
+ */
+@customElement('rh-table')
+export class RhTable extends LitElement {
+  static readonly styles = [styles];
+
+  get #table(): HTMLTableElement | undefined {
+    return this.querySelector('table') as HTMLTableElement | undefined;
+  }
+
+  get #cols(): NodeListOf<HTMLTableColElement> | undefined {
+    return this.querySelectorAll('col') as NodeListOf<HTMLTableColElement> | undefined;
+  }
+
+  get #rows(): NodeListOf<HTMLTableRowElement> | undefined {
+    return this.querySelectorAll('tbody > tr') as NodeListOf<HTMLTableRowElement> | undefined;
+  }
+
+  get #summary(): HTMLElement | undefined {
+    return this.querySelector('[slot="summary"]') as HTMLElement | undefined;
+  }
+
+  #logger = new Logger(this);
+
+  render() {
+    return html`
+      <div id="container"
+           part="container"
+           @pointerleave=${this.#onPointerleave}
+           @pointerover=${this.#onPointerover}>
+        <slot @request-sort="${this.#onRequestSort}" @slotchange=${this.#onSlotChange}></slot>
+        <slot id="summary" name="summary"></slot>
+      </div>
+    `;
+  }
+
+  #onPointerleave() {
+    if (!this.#cols) {
+      return;
+    }
+
+    this.#cols.forEach(col => col.classList.remove('active'));
+  }
+
+  #onPointerover(event: PointerEvent) {
+    if (!this.#cols) {
+      return;
+    }
+
+    event.preventDefault();
+
+    let { target } = event;
+
+    if (!(target instanceof Element)) {
+      return;
+    }
+
+    if (!['td', 'th'].includes(target.tagName)) {
+      const ancestorCell = target.closest('td, th');
+      if (ancestorCell) {
+        target = ancestorCell;
+      } else {
+        return;
+      }
+    }
+
+    this.#cols.forEach((col, index) => {
+      const { cellIndex } = target as HTMLTableCellElement;
+      if (index === cellIndex) {
+        col.classList.add('active');
+      } else {
+        col.classList.remove('active');
+      }
+    });
+  }
+
+  #onSlotChange() {
+    if (this.#table && this.#summary) {
+      this.#table.setAttribute('aria-describedby', 'summary');
+    }
+  }
+
+  #onRequestSort(event: Event) {
+    if (event instanceof RequestSortEvent) {
+      for (const button of this.querySelectorAll<RhSortButton>('rh-sort-button')) {
```

Comment:
not sure we need the type assertion on qsa here. as long as the selector is just the tagname, typescript should pick it up